scheduler/scheduler-server/src/test/java/functionaltests/workflow/TWorkflowJobs.java

```diff
@@ -155,6 +158,46 @@ public static JobId testJobSubmission(SchedulerTHelper schedulerHelper, Job jobT
 
         Assert.assertEquals(jInfo.getJobId(), id);
         Assert.assertEquals(JobStatus.RUNNING, jInfo.getStatus());
+        return id;
+    }
+
+    public static JobId waitForJobFinishedAndCompareDependencies(JobId id, SchedulerTHelper schedulerHelper,
```

Comment:
To my pov.. lack of comment in this function (to explain each step, the first "for Task t"  loop and the second one..

Reply:
Indeed, there is an original lack of comment which I did not improve. I will add some explanation

Reply:
I tried to improve comments, as requested